src/jit/morph.cpp

```diff
@@ -9190,7 +9190,7 @@ GenTree* Compiler::fgMorphRecognizeBoxNullable(GenTree* compare)
         return compare;
     }
 
-    if (opCns->gtIntConCommon.IconValue() != 0)
+    if (!opCns->IsZero())
```

Comment:
In principle this is checking a more conservative condition than required here. The reason it will not be a correctness issue is that if opCns is anything other than an GT_CNS_INT, execution will not reach here.  But the right condition to check here is

`If (!opCns->IsIntZero())`